binary-codec/codec.go

```
@@ -116,6 +125,55 @@ func EncodeForSigningClaim(json map[string]any) (string, error) {
+	}
+
+	// Create UInt32 for flags
+	flagsType := &types.UInt32{}
+	flagsBytes, err := flagsType.FromJSON(json["flags"])
+	if err != nil {
+		return "", err
+	}
+
+	// Create UInt32 for txIDs length
+	txIDsLengthType := &types.UInt32{}
+	txIDsLengthBytes, err := txIDsLengthType.FromJSON(uint32(len(txIDsInterface)))
```

golangci-lint (1.64.8)

[high] G115: integer overflow conversion int -> uint32

(gosec)